twitterscraper/tweet.py

```diff
@@ -74,3 +148,25 @@ def from_html(cls, html):
                     pass  # Incomplete info? Discard!
                 except TypeError:
                     pass  # Incomplete info? Discard!
+
+    @staticmethod
+    def extract_links_and_hashtags(soup_html):
```

Comment:
I would suggest to use @classmethod here to be consistent with the rest of the module. (using different decorators for essantially the same thing can be distracting)

Reply:
I use @staticmethod for show "**this method is independent of `instance` but related to this class**", so I think using @staticmethod is appropriate here.

However, I very well understand what you said, "using different decorators confusing".
What do you think about my opinion?

Reply:
I agree that the method should be a class method. Python provides two different decorators for declaring class methods: `@classmethod` and `@staticmethod`. The two decorators do basically the same thing: [What is the difference between @staticmethod and @classmethod?](https://stackoverflow.com/questions/136097/what-is-the-difference-between-staticmethod-and-classmethod)

The function `from_html` uses the decorator `@classmethod`. I believe that `@staticmethod` and `@classmethod` should not be mixed in the same file if there isn't a good reason for it. Mixing the two decorators can cause unnecessary confusion for the readers. 

Can you follow my line of thought and what do you think about this?

Reply:
@twollnik
I can hear you and thanks for useful link.
I really appreciate it that you give me the good oppotunity to rethink about how to write with Python. As I really don't intend to offend you, I'd like to have my opinion(understanding) about `@classmethod` and `@staticmethod` heard.

---

For readability two decarator shouldn't be used in the same file but I don't think so.
**Mixing `@classmethod` and `@staticmethod` in the same file is not bad**, or rather, unifying `@classmethod` causes misunderstaindings, I think.

We shouldn't basically consider two decorators as the same things and it is better to use them depending on the situation, though redusing readability.

I think we should use respectively:
* `@classmethod`
  * Methods that access NOT to instance methods, instance variables(attributes) BUT to class variables and class methods
  * Methods that is used as **factory method** which generates new instance(like `Tweet.from_soup`)

* `@staticmethod`
  * Utility methods that is related to class
  * Generally `@staticmethod` is basically the same as utility global functions, however, differ in terms of **if they are bound to class** or not. By using `@staticmethod`, We can EXPLICITLY say "This utility function is for a certain class".

As above, I use them. The problem that "mixing @classmethod and @staticmethod is confusing" can be solved in the source code, for instance, by sorting methods.

Feel free to tell me if you think something about my opinion:)

Reply:
Thanks for taking the time to explain your point of view in more detail. 

I think that you are making a good point. In the end it all comes down to personal preference, I guess :)